src/Type/Php/MbConvertEncodingFunctionReturnTypeExtension.php

```diff
@@ -49,6 +55,45 @@ public function getTypeFromFunctionCall(
 			return null;
 		}
 
+		if ($this->phpVersion->throwsValueErrorForInternalFunctions()) {
+			if (!isset($functionCall->getArgs()[2])) {
+				return $result;
+			}
+			$fromEncodingArgType = $scope->getType($functionCall->getArgs()[2]->value);
+
+			$mayNotDetectEncoding = false;
+			if (!$fromEncodingArgType->isArray()->no()) {
+				$constantArrays = $fromEncodingArgType->getConstantArrays();
+				if (count($constantArrays) > 0) {
+					foreach ($constantArrays as $constantArray) {
+						if (count($constantArray->getValueTypes()) > 1) {
+							$mayNotDetectEncoding = true;
```

Comment:
the name looks weird to be here assuming the condition is correct. An array with multiple values will detect encoding.

Reply:
Does `returnFalseIfCannotDetectEncoding` is better ?